src/openrct2/windows/editor_bottom_toolbar.c

```diff
@@ -271,7 +253,17 @@ void window_editor_bottom_toolbar_jump_forward_from_object_selection()
 		return;
 
 	if (gScreenFlags & SCREEN_FLAGS_TRACK_DESIGNER) {
-		sub_66F6E3();
+		for (sint32 i = 0; i < 4; i++) {
```

Comment:
It might be possible to rewrite this using existing research functions anyway (e.g. one I just created, I think).